control-plane/agents/src/bin/core/tests/rebuild/mod.rs

```diff
@@ -169,14 +176,118 @@ async fn replica_wait_and_full_rebuild() {
         .await
         .unwrap();
 
-    let volume_state = volume.state;
-    assert!(volume_state.status == VolumeStatus::Online);
+    let volume_state = volume.state.clone();
+    assert_eq!(volume_state.status, VolumeStatus::Online);
     let nexus = volume_state.target.unwrap();
     assert_eq!(nexus.children.len(), 3);
     nexus
         .children
         .iter()
-        .for_each(|c| assert!(c.state == ChildState::Online));
+        .for_each(|c| assert_eq!(c.state, ChildState::Online));
+
+    // Check where the replicas are, apart from vol target node.
+    let replicas = api_client.replicas_api().get_replicas().await.unwrap();
+    tracing::info!({?replicas}, "Available replicas - ");
+    let testrep = replicas.iter().find(|r| r.node != vol_target).unwrap();
+    let testrep_node = &testrep.node;
+    tracing::info!(
+        "Stopping node {testrep_node} having replica {}",
+        testrep.uri.clone()
+    );
+
+    // Stop the container and don't restart.
+    cluster
+        .composer()
+        .stop(&testrep_node.to_string())
+        .await
+        .expect("container stop failure");
+
+    // Check the pool on restarted node shouldn't be online.
+    // This must expire the wait timeout.
```

Comment:
Instead, why not simply wait until the pool is offline?

Reply:
We wanted to wait till timeout elapses. That needs to be done anyway for Full rebuild to start right? So we are checking throughout the interval which was 15s pool is not back online.

Reply:
Then we're waiting for the wrong thing here, we should wait until the child is rebuilt using rebuild history?

Reply:
Waiting for 500ms now. 100ms is cache period

Reply:
We will be polling for rebuild history in full rebuild case. I have also removed existing cases. So there are 2 cases which will test rebuild as well as history.